master/internal/db/group_intg_test.go

```diff
@@ -0,0 +1,246 @@
+//go:build integration
+// +build integration
+
+package db
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/determined-ai/determined/master/pkg/model"
+)
+
+func TestUserGroups(t *testing.T) {
+	ctx := context.Background()
+	pgDB := MustResolveTestPostgres(t)
+	MustMigrateTestPostgres(t, pgDB, migrationsFromDB)
+
+	t.Cleanup(func() { cleanUp(ctx, t, pgDB) })
+	setUp(ctx, t, pgDB)
+
+	t.Run("group creation", func(t *testing.T) {
+		_, err := pgDB.AddGroup(ctx, testGroup)
+		require.NoError(t, err, "failed to create group")
+	})
+
+	t.Run("search groups", func(t *testing.T) {
+		groups, err := pgDB.SearchGroups(ctx, 0)
+		require.NoError(t, err, "failed to search for groups")
+
+		index := groupsContain(groups, testGroup.ID)
+		foundGroup := groups[index]
+		require.NotEqual(t, -1, index, "Expected groups to contain the new one")
```

Comment:
if it's -1 first line will probably fail with some panic because go doesn't have negative indices
```suggestion
		require.NotEqual(t, -1, index, "Expected groups to contain the new one")
		foundGroup := groups[index]
```

Reply:
Oh, good catch. Line order should be reversed here.